python/pyspark/ml/image.py

```diff
@@ -0,0 +1,139 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.ml.util import *
+from pyspark.ml.param.shared import *
+from pyspark.sql.types import *
+from pyspark.sql.types import Row, _create_row
+from pyspark.sql import DataFrame, SparkSession, SQLContext
+import numpy as np
+
+undefinedImageType = "Undefined"
+
+imageFields = ["origin", "height", "width", "nChannels", "mode", "data"]
+
+
+def getOcvTypes(spark=None):
+    """
+    Returns the OpenCV type mapping supported
+
+    :param sparkSession (SparkSession): The current spark session
+    :rtype dict: The OpenCV type mapping supported
+
+    .. versionadded:: 2.3.0
+    """
+    spark = spark or SparkSession.builder.getOrCreate()
+    ctx = spark.sparkContext
+    return ctx._jvm.org.apache.spark.ml.image.ImageSchema.ocvTypes
```

Comment:
This looks a JVM object too:

```python
>>> spark.sparkContext._jvm.org.apache.spark.ml.image.ImageSchema.ocvTypes()
JavaObject id=o41
>>> spark.sparkContext._jvm.org.apache.spark.ml.image.ImageSchema.ocvTypes().getClass().toString()
u'class scala.collection.immutable.HashMap$HashTrieMap'
```

Reply:
Probably, we should convert this one to `java.util.HashMap()` so that automatically py4j detects this as `dict`.
Quick workaround I could think is, somewhere:

```scala
import scala.collection.JavaConverters._

val _ocvTypes: util.Map[String, Int] = ImageSchema.ocvTypes.asJava
```

Python side:

```python
spark.sparkContext._jvm.org.apache.spark.ml.image.ImageSchema._ocvTypes()
```

Reply:
Or.. just simply define a dict in Python side. ....

Reply:
nice, I will do the py4j workaround -- I originally had the extra dict on python side but then it was recommended to not have duplicate dictionaries defined in python and scala side for maintenance, which makes sense